packages/wallet-connectors/src/WalletConnect.ts

```diff
@@ -249,8 +250,15 @@ export class WalletConnectConnection implements WalletConnection {
         payload: AccountTransactionPayload,
         typedParams?: TypedSmartContractParameters
     ) {
+        // Look up transaction type name via the value in the enum 'TransactionKindString'
+        // with the same name as the provided 'AccountTransactionType' value.
+        // This is correct because the values have the same names in both enums.
+        // TypeScript isn't able to infer this, so we have to disable type checking for the lookup.
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        const typeStr = TransactionKindString[AccountTransactionType[type]];
```

Comment:
Yeah it is annoying that typescript generalizes the type to string, but I'd rather do this to avoid the error:
```suggestion
        const typeStr = TransactionKindString[AccountTransactionType[type] as keyof typeof AccountTransactionType];
```
(So that typeStr is not just `any` type)

But I can see in this case that that actually reveals a mismatch between the enums:
TransferWithScheduleAndMemo vs TransferWithScheduleWithMemo
:upside_down_face:

Reply:
Yeah just realized the same thing by writing a test for it in common-sdk... :/

But nice catch, played with that syntax but didn't get it to work. I think we'll have to write a mapper function between the two enums (as compared to handling that special case here); what do you think?

Reply:
Updated to use new SDK function.